### ISSUE TITLE
Provide an interface for users to select features they want to use instead of using arguments and how to put space in ESSID

### ISSUE THREAD

**Issue by v4l3nc3** [wifiphisher/wifiphisher#722]
Why the hell is there not an interface for Wi-Fi Pisher. Why did you use arguments instead of straight-up, organized user input!?  Users can't put a space in the SSID, which is pretty freaking important. I already fixed it myself and added an interface to make it easier, but YOU should too. Especially when the SSID is "Google Starbucks". It has a space. Am I wrong? Is there another way?

**Comment by v4l3nc3** [comment]
Phisher*

**Comment by anakin1028** [comment]
@v4l3nc3 

> Why did you use arguments instead of straight-up, organized user input

This is a good suggestion. As our features are growing, a good interface for users to select the features they want to use is straightforward for them. I'll mark this issue for `enhancement` for this reason.

>Users can't put a space in the SSID, which is pretty freaking important.

you have two ways to specify essid for now:
```
* wifiphisher --essid "Google Starbucks"  # rememeber to put the quotation marks
* wifiphisher # select the essid you want to attack on ap-selection curses section
```

**Comment by v4l3nc3** [comment]
I would definitely also like to bring up the fact that Android devices, and
Google Chrome KNOW that they are subject to a phishing attack, and they
warn the user. iPhones fall for it, but Google/Samsung products do NOT.
Lastly, you need to work on the fact that the access point requires the
ability to enter monitor mode, which is completely ridiculous. You also
can't bridge a Wi-Fi connected interface if it can't enter monitor mode.
Why? Why?! This doesn't even make any sense.

**Comment by anakin1028** [comment]
> Lastly, you need to work on the fact that the access point requires the
ability to enter monitor mode, which is completely ridiculous. 

I'm not sure I have captured what you mean. If an interface is used for AccessPoint then it cannot be used as monitor mode.

> You also can't bridge a Wi-Fi connected interface if it can't enter monitor mode.

What do you mean by this? I don't get it. Two questions for it:
* Why do you want to bridge the Wi-Fi connected interface and I guess this interface you mentioned is the AccessPoint interface?

* Why is it required to become monitor mode to do the bridge thing?

**Comment by v4l3nc3** [comment]
If my wireless card isn't in monitor mode, then WiFiPhisher won't create an
access point. I read the code, dude. Did you not read the part where it
literally EXITS if you arent in monitor mode?

On Sep 2, 2017 7:25 PM, "Anakin" <notifications@github.com> wrote:

Lastly, you need to work on the fact that the access point requires the
ability to enter monitor mode, which is completely ridiculous.

I'm not sure I have captured what you mean. If an interface is used for
AccessPoint then it cannot be used as monitor mode.

You also can't bridge a Wi-Fi connected interface if it can't enter monitor
mode.

What do you mean by this? I don't get it. Two questions for it:

   -

   Why do you want to bridge the Wi-Fi connected interface and I guess this
   interface you mentioned is the AccessPoint interface?
   -

   Why is it required to become monitor mode to do the bridge thing?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/wifiphisher/wifiphisher/issues/722#issuecomment-326780256>,
or mute the thread

**Comment by v4l3nc3** [comment]
Also, you want to give the target an internet connection once they give you
their credentials. It makes it more authentic, and it makes
Android/Samsung/Chrome devices unsuspecting of our access point. It makes
the device open it up as if it's a login. Without an internet connected
interface, the device will warn the user that they are being phished.

**Comment by anakin1028** [comment]
No. To create an access pointer the card needs not to support monitor mode.
We use `hostapd` approach and if the card support the `AP` mode it can change to `AP` from `managed` when `hostapd` is lunched

**Comment by anakin1028** [comment]
> Also, you want to give the target an internet connection once they give you
their credentials. It makes it more authentic, and it makes
Android/Samsung/Chrome devices unsuspecting of our access point. It makes
the device open it up as if it's a login. Without an internet connected
interface, the device will warn the user that they are being phished.

Yes. this feature has been mentioned in #617 and we may implement it in the future.

**Comment by v4l3nc3** [comment]
Yeah, that's bullsh*t. Try your program out with a regular wireless card
and see that you're wrong.

**Comment by anakin1028** [comment]
No. You are misunderstanding.

The wireless card must support AP mode to become a soft AP.
Indeed we can also use `monitor` mode to implement the AccessPoint by `scapy` but this will not be efficient enough for the stuffs required by AccessPoint (i.e. The beacons cannot be sent out accurate 100ms periodically since it sends from user space if we use monitor mode to implement access point).  

Another drawback is we need to implement the wheels that `hostapd` has implemented which is not reasonable. You can read the discussion in #569.

**Comment by v4l3nc3** [comment]
I'm misunderstanding? Lmao. You speak out of ignorance. I can use my
Broadcom wireless card as an access point (AP) all of the f*cking time,
whenever I want. With WIFIPHISHER, that is NOT the case. WIFIPHISHER says
that I need to enable monitor mode. Read your f*cking code before you tell
me that I'm misunderstanding. You even left a tag "#Make sure the wireless
card can enter monitor mode" or something like that (not word for word, but
I'm 99.99999% sure it's in there)

On Sep 2, 2017 7:43 PM, "Anakin" <notifications@github.com> wrote:

No. You are misunderstanding.

The wireless card must support AP mode to become a soft AP.
Indeed we can also use monitor mode to implement the AccessPoint by scapy
but this will not be efficient enough for the stuffs required by
AccessPoint (i.e. The beacons cannot be sent out accurate 100ms
periodically since it sends from user space if we use monitor mode to
implement access point).

Another drawback is we need to implement the wheels that hostapd has
implemented which is not reasonable. You can read the discussion in #569
<https://github.com/wifiphisher/wifiphisher/issues/569>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/wifiphisher/wifiphisher/issues/722#issuecomment-326780766>,
or mute the thread

**Comment by anakin1028** [comment]
can you post the output of the following:
```
iw dev
iw phy phy0 info # change the phy0 to the one you have observed in iw dev
```
I guess you have only one card and your card doesn't support the `monitor` mode.
You should use the command `wifiphihser -nJ` or `wifiphisher -nJ --essid haha`

**Comment by v4l3nc3** [comment]
I'm not talking about jamming. Jamming works fine with my wireless card
that supports monitor mode. My OTHER card doesnt support monitor mode, yet
it works fine as an AP. Dont tell me i have one card. I have several. My
point is, you shouldnt require a card that supports monitor mode when
literally ANY wireless card can support being an access point. Your program
doesnt allow a wireless card to become an access point unless it is in
monitor mode. For the last time. I READ THE CODE. It literally says that.
It exits the program if the card isnt in monitor mode, even though it
DOESNT NEED TO BE.

On Sep 2, 2017 7:51 PM, "Anakin" <notifications@github.com> wrote:

can you post the output of the following:

iw dev
iw phy phy0 info # change the phy0 to the one you have observed in iw dev

I guess you have only one card and your card doesn't support the monitor
mode.
You should use the command wifiphihser -nJ or wifiphisher -nJ --essid haha

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/wifiphisher/wifiphisher/issues/722#issuecomment-326781040>,
or mute the thread

**Comment by anakin1028** [comment]
> My point is, you shouldnt require a card that supports monitor mode when
literally ANY wireless card can support being an access point. 

I want to make sure I understand the scenario you mentioned. Do you mean the following scenario:
```
Card 1 supports AP and Managed
Card 2 doesn't support Monitor
```
And wifiphisher will require card2 to support monitor mode?

**Comment by v4l3nc3** [comment]
Let's say in this scenario, I only use ONE wireless card. That card
supports AP mode, but NOT monitor mode. I attempt to use WiFiPhisher to
make it into an access point, but NOPE. WiFi Phisher wont allow my wireless
card to become an access point because it isnt in monitor mode. Do you
understand now? FIX IT.

**Comment by anakin1028** [comment]
Wifiphisher default enable jamming process so it will require two cards. One card supports jamming and the other card supports AP.

If you have only one card, you should use the command `wifiphisher -nJ` as I said before.

**Comment by v4l3nc3** [comment]
As I said, it doesn't work.

**Comment by anakin1028** [comment]
OK. I know what you mean. We'll set it to monitor mode since we want to do the AP-selection in the recognize phase.

But if users provide the `--essid` option, we don't need to do it and we can just fork the access point.
Thanks for the suggestion.

**Comment by v4l3nc3** [comment]
Lol, mkay.

**Comment by anakin1028** [comment]
@v4l3nc3 the issue you mentioned has been solve in this branch: `fix_enabling_mon_essid_used`
you can confirm by pull that branch and we'll merge that soon. Thanks.

**Comment by v4l3nc3** [comment]
Nothing was solved.

**Comment by anakin1028** [comment]
The following command not works for you?

`wifiphisher --essid fakeAP -nJ`

**Comment by v4l3nc3** [comment]
Nope.

**Comment by sophron** [comment]
@v4l3nc3 You will have to calm down and realize that you are talking to volunteers who could be having a beer with their friends but are here helping you instead. Read our <a href="https://github.com/wifiphisher/wifiphisher/blob/master/CODE_OF_CONDUCT.md">Code of Conduct</a> and act accordingly.

**Comment by v4l3nc3** [comment]
Lmao, dude. You don't need to be helping me. I'm helping you by pointing
out the bullshit issues in your program. Major flaws that could be fixed
easily. If you want to patronize me, you can go fuck yourself accordingly.
I won't pretend to give a shit. This program is presented at HTCIA
conferences, and still has major flaws. I reccomend you fix them. That's
all, "bud".

**Comment by sophron** [comment]
@v4l3nc3 is now blocked.

@anakin1028, I removed the "enhancement" tag. Feel free to add the TUI feature selection as a new issue if you want.

**Comment by anakin1028** [comment]
sure : ) the other problem mentioned by @v4l3nc3 should be fixed in #723.

**Comment by DanMcInerney** [comment]
@sophron you have the patience of a saint. All I wanted to do was smack this idiot in the mouth after reading this thread.
